OpenSSL-Strategy.md

```diff
@@ -0,0 +1,361 @@
+it support FIPS in that configuration?
+
+The plan described above is to:
+
+* Remain backward-compatible with OpenSSL 1.0.2 via dynamic linking for the
+  lifetime of Node.js 10. This will cease being an _official_ policy at the
+  OpenSSL 1.0.2 EOL date which will occur 4 months prior to Node 10 entering
+  Maintenance LTS. However, this support will be maintained as long as it does
+  not cause security problems and there are contributors available to maintain
+  such support. The lack of a passing test suite with Node.js 10 compiled
+  against OpenSSL 1.0.2 past the EOL date will not hold up further releases of
+  Node 10.
+* Upgrade to OpenSSL 1.1.1 when made generally available and Node.js 10 can
+  retain ABI and API compatibility. OpenSSL 1.1.1 is claimed by the OpenSSL
+  project to be API and ABI compatible with 1.1.0, and Node.js experiments
+  indicates that it is, see
+  https://github.com/nodejs/node/issues/18770#issuecomment-446785733.
+* **No support for FIPS** The latest [OpenSSL 3.0 and FIPS Update][] does not
+  predict that an OpenSSL 1.1.1 ABI compatible library with FIPS support will
+  ever exist. Without such a library, Node.js 10.x will never support FIPS.
+  The lack of FIPS support is unfortunate, however, Node.js users requiring FIPS
+  support should be able to use Node.js 8 and switch to a future Node.js version
+  that supports the new FIPS module (ideally Node.js 12, but probably Node.js
+  13.x or 14.x).  See [OpenSSL 3.0.0 and FIPS][] for more information.
+
+This plan must be communicated to users of Node.js 10 early and often.  There is
+potential for instability and a change in default OpenSSL version is
+unprecedented and therefore unexpected. The potential for breaking API and/or
+ABI may also cause disruption, potentially requiring an increment of
+`NODE_MODULE_VERSION`, which will also be unprecedented within a single release
+line. There is no indication yet that this will happen when OpenSSL 1.1.1 is
+included in Node.js 10.x, but it is important that users be aware of this
+possibility.
+
+
+### Node.js version 11.x (EOL June-2019)
+
+This release will not be designated LTS. It was updated to include OpenSSL 1.1.1
+in [11.9.0](https://nodejs.org/en/blog/release/v11.9.0/).
+
+For Node.js < 11.9.0:
+* OpenSSL version: 1.1.0
+* Allowed shared OpenSSL version: 1.0.2 or 1.1.0
+* Default minimum TLS version is TLSv1, default maximum is TLSv1.2. No other
+  TLS versions are supported by this version of OpenSSL or Node.js.
+* FIPS: not supported
+
+For Node.js >= 11.9.0:
+* OpenSSL version: 1.1.0
+* Allowed shared OpenSSL version: 1.0.2 or 1.1.0 or 1.1.1
+* Default minimum TLS version is TLSv1, default maximum is TLSv1.2. It is
+  expected that TLS1.3 support will be backported to 11.x, but it will *not* be
+  supported by default, only by explicit run-time configuration.
+* FIPS: not supported
+
+XXX Confirm what versions of shared OpenSSL are expected to be supported.
```

Comment:
I will confirm what shared openssl versions we supported before 11.9.0 (when openssl 1.1.1 was integrated), and after 11.9.0.

Reply:
confirmed, 1.0.2 is not supported, 1.1.0 is